### PR TITLE
[FLINK-31339][tests] Fix unstable tests of flink-end-to-end-tests-sql module

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
## What is the purpose of the change

Fix unstable tests of flink-end-to-end-tests-sql module

## Brief change log

MINI-BATCH may exceed the latency of time, resulting result is unstable.
We made some modifications on the test, and the materialized final result was checked to avoid this problem.

Other problems caused by the same cause:
- [FLINK-31033]
- [FLINK-31141]
- [FLINK-32269]

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by Jiabao-Sun** [comment]
Hi @XComp, could you please help review this when you have time?

**Comment by XComp** [comment]
That's not my field of expertise. Could you ping someone else who is better in judging whether the change makes sense.

**Comment by Jiabao-Sun** [comment]
Okay, thanks @XComp.
@PatrickRen @lsyldliu, could you help review this when you have time?

**Comment by Jiabao-Sun** [comment]
@luoyuxia, @lsyldliu, please take a look when you have time.

**Comment by Jiabao-Sun** [comment]
Hi @leonardBang, could you please help with this PR?
Thanks a lot.

**Review by leonardBang — CHANGES_REQUESTED** [review]
Thanks @Jiabao-Sun for the contribution, current change would lead to we missing the `mini-batch` and `aggregate key multiple times`  coverages,  I‘m slightly -1 about current fix.
I think we can improve the checkResult logic here https://github.com/apache/flink/blob/6ebd4dc3e4f902701d64d3709d47d971f1ac4753/flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java#L159 to fix unstable tests. e.g.: we can compare multiple times when output result sizes equals util we get expected result content or timeout.

**Comment by ruanhang1993** [comment]
Hi, @Jiabao-Sun . 
If we drop the mini-batch setting, are there still some tests testing the mini-batch feature for these sqls?

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java [hunk 1]
```diff
@@ -190,4 +231,76 @@ private static List<String> readResultFiles(Path path) throws IOException {
...
+            ResolvedSchema schema) {
+        return new DebeziumJsonDeserializationSchema(
+                schema.toPhysicalRowDataType(),
+                Collections.emptyList(),
+                InternalTypeInfo.of(schema.toPhysicalRowDataType().getLogicalType()),
+                false,
+                true,
+                TimestampFormat.ISO_8601);
+    }
+
```
Comment (ruanhang1993) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java:293]: add some java docs

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java [hunk 2]
```diff
@@ -190,4 +231,76 @@ private static List<String> readResultFiles(Path path) throws IOException {
...
+            RowKind kind = rowData.getRowKind();
+
+            Row row = converter.toExternal(rowData);
+            assertThat(row).isNotNull();
+
+            Row key = Row.project(row, schema.getPrimaryKeyIndexes());
+            key.setKind(RowKind.INSERT);
+
+            Row upsertRow = Row.copy(row);
+            upsertRow.setKind(RowKind.INSERT);
```
Comment (ruanhang1993) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java:266]: `upsertRow` is only useful when the kind is insert or update_after. So we could create it when `kind == RowKind.INSERT || kind == RowKind.UPDATE_AFTER`.

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java [hunk 3]
```diff
@@ -190,4 +231,76 @@ private static List<String> readResultFiles(Path path) throws IOException {
         }
         return result;
     }
+
+    protected List<String> formatRawResult(List<String> rawResults) {
```
Comment (ruanhang1993) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SqlITCaseBase.java:235]: Add some docs for this method on how to use or override it.

**Review by ruanhang1993 — APPROVED** [review]
Thanks for the work. LGTM.

**Comment by leonardBang** [comment]
Thanks @Jiabao-Sun and @ruanhang1993 for the contribution , could you change the PR description? IIUC, it has been outdated.

**Comment by Jiabao-Sun** [comment]
> Thanks @Jiabao-Sun and @ruanhang1993 for the contribution , could you change the PR description? IIUC, it has been outdated.

Thanks @leonardBang and @ruanhang1993 for the review.
The PR description has been updated.

**Review by leonardBang — APPROVED** [review]
+1

**Pull request merged**